packages/client/src/client.js

```diff
@@ -31,19 +31,27 @@ function validateProjectPath(path) {
 export default class PercyClient {
   log = logger('client');
   env = new PercyEnvironment(process.env);
+  clientInfo = new Set();
+  environmentInfo = new Set();
 
   constructor({
     // read or write token, defaults to PERCY_TOKEN environment variable
     token,
     // initial user agent info
-    clientInfo = '',
-    environmentInfo = '',
+    clientInfo,
+    environmentInfo,
     // versioned api url
     apiUrl = PERCY_CLIENT_API_URL
   } = {}) {
     Object.assign(this, { token, apiUrl });
-    this.clientInfo = new Set([].concat(clientInfo));
-    this.environmentInfo = new Set([].concat(environmentInfo));
+    // set our env info
+    this.addClientInfo(`Percy/${/\w+$/.exec(this.apiUrl)}`);
+    this.addClientInfo(`${pkg.name}/${pkg.version}`);
+    this.addEnvironmentInfo(`node/${process.version}`);
+    this.addEnvironmentInfo(this.env.info);
+    // set info passed from the SDK
+    this.addClientInfo([].concat(clientInfo).filter(Boolean));
+    this.addEnvironmentInfo([].concat(environmentInfo).filter(Boolean));
```

Comment:
With my other review comment below, that check wouldn't work since we're adding our own info as soon as possible here. We'll have to go back to adding this all lazily when the UA is created. To avoid duplicates (which was the goal of moving these here), we can duplicate the sets immutably with our information appended first.